SaltSquareIO/sports-app-react-fe
# Setup linter and formatter

[CLICKUP_TICKET_ID](https://app.clickup.com/t/86c0fy8wy)

## Summary of change

This pull request implements linting and formatting for the project to ensure consistent code quality and adherence to TypeScript standards. The setup includes the following features: 

- Configured linting and formatting rules appropriate for TypeScript.
- Introduced npm commands to run the linter and formatter.
- Added a precommit hook that runs the linter and formatter before allowing any commits.
- Prevents code from being committed unless it passes both the linter and formatter checks.

## QA Notes

To test this PR, follow these steps:
1. Run npm install to ensure all dependencies are installed.
2. Make a code change that violates the linting rules (e.g., incorrect spacing).
3. Attempt to commit the changes. The precommit script should prevent the commit and display messages indicating the linting and formatting process.
4. Ensure that making valid changes allows the commit to proceed.

If the linting and formatting rules are triggered correctly, the PR is functioning as intended.
